rpc/common/moc_common_networkcommon.proto

```diff
@@ -51,4 +51,10 @@ message Route {
 	string nextHop = 1;
 	string destinationPrefix = 2;
 	uint32 metric = 3;
+}
+
+message ResourceReference {
```

Comment:
I would suggest to be more specific here. Lets use NSG resource reference. 

Follow up question ,do we need a resource reference for NSG since I assume they cant be used across location. So only name is required.

Reply:
this is derived from what azure uses, for eg; subresource. we want to have a common reference type to use for various resources, not only nsg.

Reply:
Do we have the azure-sdk reference?

Reply:
also to the follow up question, moc has the pattern of using just string name, which imho is confusing.  it makes it difficult to fully qualify resources, say for eg; in case of multi-location if that comes in. so imho, its better to use a explicit resourceref. we also have a bunch work coming in slb etc. where the same pattern will be used.

Reply:
> azure-sdk

https://github.com/Azure/azure-sdk-for-go/blob/main/sdk/resourcemanager/network/armnetwork/models.go

Reply:
This is an example I am looking at. https://github.com/Azure/azure-sdk-for-go/blob/main/sdk/resourcemanager/network/armnetwork/models.go#L11181

Reply:
We are looking at the sdk models and trying to bring that to the rpc models.

Reply:
I agree with a reference, but I am looking for specific reference and not a generic one

Reply:
yeah, we wanted to use this pattern, but then it will cause a cyclic dependency in protobuf which it doesnt support. subnet will have ref to nsg, and nsg will have ref to subnet(when the back ref is introduced, this is now missing).

Reply:
We can limit the reference to have name and location . We can ignore the other properties

Reply:
> I agree with a reference, but I am looking for specific reference and not a generic one

all resource references are just stubs, whether its generic or not. we have a generic base one(this is like the inheritance pattern, but protobuf doesnt support it). then for specific scenarios if the generic one doesnt suffice introduce specific ones. but i think 90%(in networking) that we are planning to bring just need a generic stub. what i am trying to avoid is bring a lot different stubs for each scenario.